tests/cli/test_rasa_data.py

```diff
@@ -246,6 +263,7 @@ def test_validate_files_form_not_found_invalid_domain(
     )
     args = {
         "domain": "data/test_restaurantbot/domain.yml",
+        "fail_on_warnings": False,
         "data": [file_name],
         "max_history": None,
```

Comment:
Why is this flag needed?

Reply:
Addressed [here](https://github.com/RasaHQ/rasa/pull/12549#discussion_r1242114051), and also because validator used to fail fast eg `func_1 and func_2` will not run `func_2` if `func_1` returned `False` so in the tests it never used the `fail_on_warnings` until I called the functions before the `and`